### PR TITLE
PHP 8.1 support

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

This PR introduces support for PHP 8.1 as far as I was able to test that locally.
This also removes the usage of `prophecy` within this component along with calling the deprecated (and yet useless) method mysqli#init as stated in the related [RFC](https://wiki.php.net/rfc/deprecations_php_8_1#mysqliinit)

### CODE REVIEW

**Comment by boesing** [comment]
I did not fixed all the deprecation messages from phpunit as I dont want to spend more time on this than necessary.

Tests are passing, I need these changes for `laminas-session` PHP 8.1 support.

File: phpunit.xml.dist [hunk 1]
```diff
@@ -3,7 +3,9 @@
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
     bootstrap="vendor/autoload.php"
-    colors="true">
+    colors="true"
+    beStrictAboutTestsThatDoNotTestAnything="false"
```
Comment (Ocramius) [inline phpunit.xml.dist:7]: Ugly but indeed we do have some of those :-\

File: src/Adapter/Driver/Mysqli/Connection.php [hunk 2]
```diff
@@ -114,7 +114,6 @@ public function connect()
         $cipher = (isset($p['cipher'])) ? $p['cipher'] : '';
 
         $this->resource = $this->createResource();
-        $this->resource->init();
```
Comment (Ocramius) [inline src/Adapter/Driver/Mysqli/Connection.php:117]: This smells a bit 🤔

File: test/unit/Adapter/Driver/Mysqli/ConnectionTest.php [hunk 3]
```diff
@@ -133,8 +133,6 @@ public function testConnectionFails()
     protected function createMockMysqli($flags)
     {
         $mysqli = $this->getMockBuilder('\mysqli')->getMock();
-        $mysqli->expects($this->once())
-            ->method('init');
```
Comment (Ocramius) [inline test/unit/Adapter/Driver/Mysqli/ConnectionTest.php:137]: `init` does not seem to be deprecated: we should probably keep this?

File: test/unit/Adapter/Driver/Pdo/ConnectionTransactionsTest.php [hunk 4]
```diff
@@ -14,16 +14,14 @@
...
      */
     protected $wrapper;
 
-    /**
-     * {@inheritDoc}
-     */
     protected function setUp(): void
     {
         $this->wrapper = new ConnectionWrapper();
+        parent::setUp();
```
Comment (Ocramius) [inline test/unit/Adapter/Driver/Pdo/ConnectionTransactionsTest.php:24]: `parent::setUp()` should come first

File: test/unit/TestAsset/ConnectionWrapper.php [hunk 5]
```diff
@@ -12,6 +12,7 @@ class ConnectionWrapper extends Connection
     public function __construct()
     {
         $this->resource = new PdoStubDriver('foo', 'bar', 'baz');
+        parent::__construct();
```
Comment (Ocramius) [inline test/unit/TestAsset/ConnectionWrapper.php:15]: Parent ctor should be called first

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Adapter/Driver/Mysqli/Connection.php (hunk 2 above)
Comment (boesing) [reply src/Adapter/Driver/Mysqli/Connection.php:117]: https://wiki.php.net/rfc/deprecations_php_8_1#mysqliinit

File: test/unit/Adapter/Driver/Mysqli/ConnectionTest.php (hunk 3 above)
Comment (boesing) [reply test/unit/Adapter/Driver/Mysqli/ConnectionTest.php:137]: it is.
https://wiki.php.net/rfc/deprecations_php_8_1#mysqliinit

you voted for it ;-)

Comment (Ocramius) [reply test/unit/Adapter/Driver/Mysqli/ConnectionTest.php:137]: Indeed, buried under the ginormous RFC, sorry 🤣

**Comment by weierophinney** [comment]
I did the crazy work that @boesing referenced as "too much time" in #205 (and yes, it **was** too much time! going on 7 hours now!)

**Pull request closed without merging**